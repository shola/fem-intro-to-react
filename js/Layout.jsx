// @flow
import React from 'react';

type Props = {
    children: React.Element<any>
};

/**
 * Common UI between all layouts
 */
const Layout = (props: Props) => (
    <div className="app-container">
        <h1>Welcome to my video app!</h1>
        {props.children}
    </div>
);

export default Layout;
