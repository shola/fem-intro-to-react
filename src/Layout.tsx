import * as React from 'react';

interface Props {
    children: any
}

/**
 * Common UI between all layouts
 */
const Layout = (props: Props) => (
    <div className="app-container">
        {props.children}
    </div>
);

export default Layout;
