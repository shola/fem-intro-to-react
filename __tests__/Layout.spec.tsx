import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as Renderer from 'react-test-renderer';
import Layout from '../src/Layout';

xdescribe('Reusable layout component', () => {
    it('Check snapshot for unexpected changes', () => {
        const component = Renderer.create(<Layout />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
