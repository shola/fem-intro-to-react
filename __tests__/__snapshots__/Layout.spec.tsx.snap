// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reusable layout component Check snapshot for unexpected changes 1`] = `
<div
  className="app-container"
/>
`;
